kratos/processes/apply_periodic_boundary_condition_process.h

```diff
@@ -0,0 +1,196 @@
+     * @param rModelPart The main model part where the boundary conditions are applied.
+     * @param rParameters parameters for the periodic condition to be applied
+     */
+    ApplyPeriodicConditionProcess(ModelPart &rMasterModelPart, ModelPart &rSlaveModelPart,
+                                  Parameters rParameters);
+
+    /**
+     * @brief Destructor of the process class
+     */
+    ~ApplyPeriodicConditionProcess();
+
+    /**
+     * @brief Function initializes the process
+     */
+    void ExecuteInitialize() override;
+
+    /**
+     * @brief Function initializes the solution step
+     */
+    void ExecuteInitializeSolutionStep() override;
+
+    /**
+     * @brief Function to print the information about this current process
+     */
+    void PrintInfo(std::ostream& rOStream) const override;
+
+  private:
+    MatrixType mTransformationMatrix; // This can be either for rotating or for translating the slave geometry to Master geometry
+    MatrixType mTransformationMatrixVariable; // This can be either for rotating or for translating the master variable to slave geometry
+    ModelPart &mrMasterModelPart;       // the master modelpart to which the master-slave constraints are added.
+    ModelPart &mrSlaveModelPart;
+    Parameters mParameters;          // parameters
+    double mTheta;
+    bool mIsInitialized;
+    std::vector<double> mCenterOfRotation;
+    std::vector<double> mAxisOfRotationVector;
+    std::string mType;
+    double mModulus;
+    std::vector<double> mDirOfTranslation;
+
+    /**
+     * @brief  Function to remove the common nodes of slave and master modelparts from the slave modelpart
+     */
+    void RemoveCommonNodesFromSlaveModelPart();
+
+    /**
+     * @brief   The function to figure out how the master and slave model parts relate together and add master-slave constraints
+     *          to the rModelPart
+     */
+    template <int TDim>
+    void ApplyConstraintsForPeriodicConditions();
+
+    /**
+     * @brief   The function add the linear master-slave constraint to rModelPart. This function is specifically for applying
+     *          periodic conditions for vector variable. This distinction is because, for vector variables the variable should also be
+     *          transformed according to the transfromation specified in the settings.
+     * @param rSalveNode The slave node which is to be connected to the rHostGeometry.
+     * @param rHostedGeometry the Host geometry which has the rSlaveNode.
+     * @param rWeights The weights with which the rSlaveNode is connected to the rHostedGeometry's nodes.
+     * @param rVarName The name of the vector variable on which periodic boundary condition can be applied.
+     */
+    template <int TDim>
+    void ConstraintSlaveNodeWithConditionForVectorVariable(NodeType& rSalveNode, const GeometryType& rHostedGeometry, const VectorType& rWeights, const std::string& rVarName );
```

Comment:
but I still have to check if the variable exists and it have components ...

Reply:
Ok, it looks like a problem since you really want the components and not the variable. But if you can check something at compile time rather than run time, it is better.